test/unit/util.js

```diff
@@ -985,6 +1011,55 @@
     assert.deepEqual(matrix, fabric.iMatrix, 'default is identity matrix');
   });
 
+  QUnit.test('decodeTransformMatrix with angle', function(assert) {
+    assert.ok(typeof fabric.util.decodeTransformMatrix === 'function');
+    const maxDiffAllowed = 0.000000009;
+    for (let angle = 0; angle <= 360; angle += 30) {
+      [[1, 1], [1, 4], [5, 1], [3, 8]].forEach(([scaleX, scaleY]) => {
+        for (let skewX = -80; skewX < 90; skewX += 20) {
+          for (let skewY = -80; skewY < 90; skewY += 20) {
+            [
+              [false, false], 
+              [true, false], 
+              [false, true], 
+              [true, true]
+            ].forEach(([flipX, flipY]) => {
```

Comment:
This nested loop works on node but gives weird error in firefox and chrome...

Firefox:
```
Stdout: 
             [GFX1-]: RenderCompositorSWGL failed mapping default framebuffer, no dt
```

It only works when I run fewer test cases. It seems to be some memory overflow.

I didn't find much about this error, just this here: https://bugzilla.mozilla.org/show_bug.cgi?id=1716303. It seems to me to be a log spamming bug. Maybe this causes memory overflow(?)

Reply:
I fixed it here https://github.com/fabricjs/fabric.js/pull/8425/commits/6353f9d27e0b6425a29664a6a575758f7ab01d82 by separating it into several test cases. It was kind of ugly, but it was the way I managed to make it run in firefox and chrome without error.

Reply:
I am experiencing this on my machine running in firefox
Can you post the command you are running?

Reply:
I see the logs on CI....
I have no idea why and what

Reply:
npm run test -- -c chrome

Reply:
I did it in a way that looks a little better. I'll commit

Reply:
https://github.com/fabricjs/fabric.js/pull/8425/commits/9d8921a9113d8700d1776036743d58f03422cc89 
I adapted from your commit. The problem seems to be test cases that are too large.. Locally ran without problems